cmd/ipfs/kubo/daemon.go

```diff
@@ -704,6 +706,24 @@ take effect.
 	return errs
 }
 
+// TODO: should a  version of this live in https://github.com/multiformats/go-multiaddr
+// so we dont need to duplicate code here and in client/rpc/api.go ?
+func NormalizeUnixMultiaddr(address string) string {
+	// Support legacy and modern /unix addrs
+	// https://github.com/multiformats/multiaddr/pull/174
+	socketPath, err := url.PathUnescape(address)
+	if err != nil {
+		return address // nil, fmt.Errorf("failed to unescape /unix socket path: %w", err)
+	}
+	// Ensure the path is absolute
+	if !strings.HasPrefix(socketPath, string(filepath.Separator)) {
+		socketPath = string(filepath.Separator) + socketPath
```

Comment:
Is it correct to prepend a separator, or should the current directory be prepended?

See [`filepath.Abs`](https://pkg.go.dev/path/filepath@go1.24.5#Abs)